airflow/utils/python_virtualenv.py

```diff
@@ -35,12 +37,35 @@ def _generate_virtualenv_cmd(tmp_dir: str, python_bin: str, system_site_packages
     return cmd
 
 
-def _generate_pip_install_cmd(tmp_dir: str, requirements: List[str]) -> Optional[List[str]]:
+def _generate_pip_install_cmd(tmp_dir: str,
+                              requirements: List[str],
+                              connection_id: Optional[str] = None
+                              ) -> Optional[List[str]]:
     if not requirements:
         return None
-    # direct path alleviates need to activate
-    cmd = [f'{tmp_dir}/bin/pip', 'install']
-    return cmd + requirements
+
+    if connection_id:
+        con: Connection = BaseHook.get_connection(connection_id)
+        user = con.login
+        schema = con.schema or 'http'
+        password = con.get_password()
+        port = con.port
+        host = con.host
+        if user:
+            index_url = f"{schema}://{user}:{password}@{host}:{port}/repository/python/simple"
```

Comment:
There are few problems here:
* port is Optional [int] - it might be None - we need to provide sensible default if we want to hard-code : (or  handle None) case.
* "/repository/python/simple should not be hard-coded. I imagine registries can have different path. If anything the "/repository/python/simple" might be a default, but it should be overridable via extra in connection
* `https://` should be default for private registries. There are good reasons for private registry to be private and I cannot imagine a case where it would be http-based, leaving it essentially unprotected and with user/passwords sent in plain text.